lib/rouge/lexers/http.rb

```diff
@@ -32,8 +32,8 @@ def guess_content_lexer
         # request
         rule %r(
           (#{HTTP.http_methods.join('|')})([ ]+) # method
-          ([^ ]+)([ ]+)                          # path
-          (HTTPS?)(/)(\d(?:\.\d)?)(\r?\n|$)      # http version
+          ([^ ]+)(?:([ ]+)                       # path
+          (HTTPS?)(/)(\d(?:\.\d)?))?(\r?\n|$)    # http version
```

Comment:
**suggestion:** Spreading the regex across the two components make it a bit harder to read. Would it be possible to leave it as it is and remove the `push :headers` statement in both the `request` and `response` block?

What do you think about the following diff on master?

```diff
diff --git lib/rouge/lexers/http.rb lib/rouge/lexers/http.rb
index 8e4b0fbf..b28124a2 100644
--- lib/rouge/lexers/http.rb
+++ lib/rouge/lexers/http.rb
@@ -40,8 +40,6 @@ module Rouge
             Name::Namespace, Text,
             Keyword, Operator, Num, Text
           )
-
-          push :headers
         end
 
         # response
@@ -55,8 +53,9 @@ module Rouge
             Num, Text,
             Name::Exception, Text
           )
-          push :headers
         end
+
+        mixin :headers
       end
 
       state :headers do
```